testing/events.go

```diff
@@ -120,6 +120,22 @@ func ParsePacketFromEvents(events []abci.Event) (channeltypes.Packet, error) {
 	return channeltypes.Packet{}, fmt.Errorf("acknowledgement event attribute not found")
 }
 
+// ParsePacketsFromEvents parses events emitted from a MsgRecvPacket and returns
+// all the packets found.
+func ParsePacketsFromEvents(events []abci.Event) ([]channeltypes.Packet, error) {
+	var packets []channeltypes.Packet
+	for i, ev := range events {
+		if ev.Type == channeltypes.EventTypeSendPacket {
+			packet, err := ParsePacketFromEvents(events[i:])
```

Comment:
```suggestion
			packet, err := ParsePacketFromEvents(events[i:i+1])
```
Is there any reason why we are not doing this?

Reply:
True, because it you use `packet, err := ParsePacketFromEvents(events[i:])` and you have a send packet event in index, let's say 4, then you would parse the same packet multiple times, right?

Reply:
No because of the condition just above. Basically here, `events[i]` is always of type `EventTypeSendPacket`. Anyway I resolved this conversation because the code is outdated: the logic has been moved to an other function.